mbta/routes.go

```diff
@@ -0,0 +1,54 @@
+package mbta
+
+import (
+	"context"
+	"fmt"
+)
+
+const routesAPIPath = "/routes"
+
+// RouteService handling all of the route related API calls
+type RouteService service
+
+// GetAllRoutes returns all routes from the mbta API
+func (s *RouteService) GetAllRoutes(config GetAllRoutesRequestConfig) ([]*Route, error) {
+	return s.GetAllRoutesWithContext(context.Background(), config)
+}
+
+// GetAllRoutesWithContext returns all routes from the mbta API given a context
+func (s *RouteService) GetAllRoutesWithContext(ctx context.Context, config GetAllRoutesRequestConfig) ([]*Route, error) {
+	req, err := s.client.newGETRequest(routesAPIPath)
+	config.addHTTPParamsToRequest(req)
+	req = req.WithContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	untypedRoutes, _, err := s.client.doManyPayload(req, Route{})
```

Comment:
```suggestion
	untypedRoutes, _, err := s.client.doManyPayload(req, &Route{})
```

I had a mistake on master when you branched I think. This should be a pointer. I fixed it on master for the other types.